cmd/oras/internal/option/cache_test.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package option
+
+import (
+	"os"
+	"reflect"
+	"testing"
+
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/content/memory"
+	"oras.land/oras-go/v2/content/oci"
+	"oras.land/oras/internal/cache"
+)
+
+var mockTarget oras.ReadOnlyTarget = memory.New()
+
+func TestCache_CachedTarget(t *testing.T) {
+	tempDir := t.TempDir()
+	os.Setenv("ORAS_CACHE", tempDir)
+	opts := Cache{}
+
+	ociStore, err := oci.New(tempDir)
+	if err != nil {
+		t.Fatal("error calling oci.New(), error =", err)
+	}
+	want := cache.New(mockTarget, ociStore)
```

Comment:
I was trying to write a better test here, but I didn't find a good way unless exporting the `cache.target` [struct](https://github.com/oras-project/oras/blob/main/internal/cache/target.go#L36).
```
want := struct {
	oras.ReadOnlyTarget
	cache content.Storage
}{
	mockTarget,
	ociStore,
}
```
If I built a struct like above, `want` and `got` will not be equal.